pkg/util/kubernetes/kubernetes.go

```diff
@@ -145,12 +147,129 @@ func addTLSConfig(sconfig *natsconf.ServerConfig, cs spec.ClusterSpec) {
 	}
 }
 
-// addAuthConfig fills the Auth configuration to be used in config map.
-func addAuthConfig(kubecli corev1client.CoreV1Interface, ns string, sconfig *natsconf.ServerConfig, cs spec.ClusterSpec) error {
+func addAuthConfig(
+	kubecli corev1client.CoreV1Interface,
+	operatorcli natsalphav2client.PkgSpecInterface,
+	ns string,
+	clusterName string,
+	sconfig *natsconf.ServerConfig,
+	cs spec.ClusterSpec,
+	owner metav1.OwnerReference,
+) error {
 	if cs.Auth == nil {
 		return nil
 	}
-	if cs.Auth.ClientsAuthSecret != "" {
+
+	if cs.Auth.EnableServiceAccounts {
+		roleSelector := map[string]string{
```

Comment:
gometalinter complains that this function is too complex...   Maybe break it up a bit to improve readability?